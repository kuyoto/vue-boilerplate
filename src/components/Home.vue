<template>
  <v-container>
    <v-layout text-center wrap>
      <v-flex xs12>
        <v-img
          :src="require('../assets/icons/logo.svg')"
          alt="vue logo"
          class="my-3"
          contain
          height="200"
        ></v-img>
      </v-flex>

      <v-flex mb-4>
        <h1 class="display-1 font-weight-bold mb-3">{{ message }}</h1>

        <p class="subheading font-weight-regular"
          >For guide and recipes on how to configure / customize this
          project.<br />checkout the
          <a href="https://cli.vuejs.org'" target="_blank">vue-cli docs</a>
        </p>
      </v-flex>

      <v-flex xs12 mb-5>
        <h2 class="headline font-weight-bold mb-3">Installed CLI Plugins</h2>

        <v-layout justify-center>
          <a
            v-for="(plugin, i) in installedPlugins"
            :key="i"
            :href="plugin.href"
            class="subheading mx-3"
            target="_blank"
            >{{ plugin.text }}</a
          >
        </v-layout>
      </v-flex>

      <v-flex xs12 mb-5>
        <h2 class="headline font-weight-bold mb-3">Essential Links</h2>

        <v-layout justify-center>
          <a
            v-for="(link, i) in essentialLinks"
            :key="i"
            :href="link.href"
            class="subheading mx-3"
            target="_blank"
            >{{ link.text }}</a
          >
        </v-layout>
      </v-flex>

      <v-flex xs12 mb-5>
        <h2 class="headline font-weight-bold mb-3">Ecosystem</h2>

        <v-layout justify-center>
          <a
            v-for="(eco, i) in ecosystem"
            :key="i"
            :href="eco.href"
            class="subheading mx-3"
            target="_blank"
            >{{ eco.text }}</a
          >
        </v-layout>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  name: 'home',
  components: {},
  data: () => ({
    message: 'Welcome to your Vue.js + Vuetify App',
    ecosystem: [
      {
        text: 'vue-router',
        href: 'hhttps://router.vuejs.org',
      },
      {
        text: 'vue-loader',
        href: 'https://vue-loader.vuejs.org',
      },
      {
        text: 'vue-cli',
        href: 'https://cli.vuejs.org',
      },
      {
        text: 'vuex',
        href: 'ttps://vuex.vuejs.or',
      },
      {
        text: 'awesome-vue',
        href: 'https://github.com/vuejs/awesome-vue',
      },
    ],
    essentialLinks: [
      {
        text: 'Core Docs',
        href: 'https://vuejs.org',
      },
      {
        text: 'Forum',
        href: 'https://forum.vuejs.org',
      },
      {
        text: 'Community Chat',
        href: 'https://chat.vuejs.org',
      },
      {
        text: 'Twitter',
        href: 'https://twitter.com/vuejs',
      },
    ],
    installedPlugins: [
      {
        text: 'Babel',
        href: 'https://cli.vuejs.org/core-plugins/babel.html',
      },
      {
        text: 'ESLint',
        href: 'https://cli.vuejs.org/core-plugins/eslint.htm',
      },
      {
        text: 'Jest',
        href: 'https://cli.vuejs.org/core-plugins/unit-jest.html',
      },
      {
        text: 'Vuetify',
        href: 'https://awesomejs.dev/for/vue-cli/pkg/245665376837304832/',
      },
    ],
  }),
}
</script>

<style scoped>
a {
  color: #42b983;
}
</style>
