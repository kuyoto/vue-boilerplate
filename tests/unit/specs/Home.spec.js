import Vue from 'vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'
import Home from '@/components/Home.vue'

Vue.use(Vuetify)

const localVue = createLocalVue()

describe('@views/home', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should render correct and matches snapshot', () => {
    const wrapper = mount(Home, { localVue, vuetify })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toMatch('Welcome')
  })
})
