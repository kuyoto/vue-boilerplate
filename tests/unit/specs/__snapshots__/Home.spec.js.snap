// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@views/home should render correct and matches snapshot 1`] = `
<div class="container">
  <div class="layout text-center wrap">
    <div class="flex xs12">
      <div aria-label="vuetify logo" role="img" class="v-image v-responsive my-3 theme--light" style="height: 200px;">
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="flex mb-4">
      <h1 class="display-2 font-weight-bold mb-3">Welcome to Vuetify</h1>
      <p class="subheading font-weight-regular">For help and collaboration with other vuetifydevlopers, <br>please
        join our online
        <a href="https://community.vuetifyjs.com" target="_blank">Discord community</a></p>
    </div>
    <div class="flex xs12 mb-5">
      <h2 class="headline font-weight-bold mb-3">What's next?</h2>
      <div class="layout justify-center"><a href="https://vuetifyjs.com/components/api-explorer" target="_blank" class="subheading mx-3">Explore components</a><a href="https://vuetifyjs.com/layout/pre-defined" target="_blank" class="subheading mx-3">Select a layout</a><a href="https://vuetifyjs.com/getting-started/frequently" target="_blank" class="subheading mx-3">FAQ</a></div>
    </div>
    <div class="flex xs12 mb-5">
      <h2 class="headline font-weight-bold mb-3">Essential Links</h2>
      <div class="layout justify-center"><a href="https://vuetifyjs.com" target="_blank" class="subheading mx-3">Documentation</a><a href="https://community/vuetifyjs.com" target="_blank" class="subheading mx-3">Chat</a><a href="https://madewithvuejs/vuetify" target="_blank" class="subheading mx-3">Made with Vuetify</a><a href="https://twitter.com/vuetifyjs" target="_blank" class="subheading mx-3">Twitter</a><a href="https://medium.com/vuetify" target="_blank" class="subheading mx-3">Articles</a></div>
    </div>
    <div class="flex xs12 mb-5">
      <h2 class="headline font-weight-bold mb-3">Ecosystem</h2>
      <div class="layout justify-center"><a href="https://github.com/vuetifyjs/vuetifyjs-loader" target="_blank" class="subheading mx-3">vuetify-loader</a><a href="https://github.com/vuetifyjs/vuetifyjs" target="_blank" class="subheading mx-3">github</a><a href="https://github.com/vuetifyjs/awesome-vuetifyjs" target="_blank" class="subheading mx-3">awesome-vuetify</a></div>
    </div>
  </div>
</div>
`;
